jekyll/_cci2/configuration-reference.md

```diff
@@ -1391,6 +1391,139 @@ ignore | N | String, or List of Strings | Either a single tag specifier, or a li
+exclude | N | Map | A list of argument maps that should be excluded from the matrix
+alias | N | String | An alias for the matrix, usable from another job's `requires` stanza. Defaults to the name of the job being executed
+
+The following is a basic example of using matrix jobs. All features are
+explained in more detail below.
+
+```yaml
+workflows:
+  workflow:
+    jobs:
+      - build:
+        matrix:
+          parameters:
+            version: ["0.1", "0.2", "0.3"]
+            platform: ["macos", "windows", "linux"]
+```
+
+###### Excluding sets of parameters from a matrix
+{:.no_toc}
+Sometimes you may wish to run a job with every combination of arguments _except_
+some value or values. You can use an `exclude` stanza to achieve this:
+
+```yaml
+workflows:
+  workflow:
+    jobs:
+      - build:
+          matrix:
+            parameters:
+              a: [1, 2, 3]
+              b: [4, 5, 6]
+            exclude:
+              - a: 3
+                b: 5
+```
+
+The matrix above would result in 8 jobs being run: every combination of the
+parameters `a` and `b`, excluding `{a: 3, b: 5}`
+
+###### Requiring an entire matrix
+{:.no_toc}
+Another job can `require` an entire matrix (that is, every job in the matrix)
+using its `alias`. The `alias` defaults to the name of the job being invoked.
+```yaml
+workflows:
+  workflow:
+    jobs:
+      - deploy:
+          matrix:
+            parameters:
+              version: ["0.1", "0.2"]
+      - another-job:
+          requires:
+            - deploy
+```
+
+This means that `another-job` will require both `deploy-0.1` and `deploy-0.2` to
+finish before it runs. 
+
+You can also manually set the `alias` to a custom value. For example:
+
+```yaml
+workflows:
+  workflow:
+    jobs:
+      - deploy:
+          matrix:
+            parameters:
+              version: ["0.1", "0.2"]
+            # setting a custom alias
+            alias: my-deploy-matrix
+      - another-job:
+          requires:
+            # using a custom alias
+            - my-deploy-matrix
+```
+
+###### Matrix parameters
+{:.no_toc}
+When running a job in a matrix, other job arguments (such as `requires` or
+`name`) can contain matrix parameters, in the form `<< matrix.[parameter-name] >>`
+For example:
+```yaml
+workflows:
+  workflow:
+    jobs:
+      - build-0.1: ...
+      - build-0.2: ...
+      - deploy:
+          matrix:
+            parameters:
+              version: ["0.1", "0.2"]
+          name: "deploy-<< matrix.version >>"
+          requires:
+            - "build-<< matrix.version >>"
+```
+
+This matrix results in two jobs being run. One, called `deploy-0.1`,
+would require the `build-0.1` job. The other, called `deploy-0.2`, would require
+the `build-0.2` job.
+
+In a typical use case, you may wish to require a single job from a matrix from
+within another matrix. To do so, use matrix parameters to set the `name` of the
+required job, and the `requires` stanza of the dependent job:
```

Comment:
Idea: "requiring an entire matrix" and "requiring matrix jobs from within a matrix" feel like similar enough topics that they could be grouped together into a section called `Dependencies and matrix jobs` that showcases the two use cases people might be interested in.